flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java

```diff
@@ -107,16 +101,69 @@ public SchemaRegistryRequestHandler(
             SchemaDerivation schemaDerivation,
             SchemaChangeBehavior schemaChangeBehavior) {
         this.metadataApplier = metadataApplier;
-        this.activeSinkWriters = new HashSet<>();
-        this.flushedSinkWriters = new HashSet<>();
-        this.pendingSchemaChanges = new LinkedList<>();
-        this.finishedSchemaChanges = new LinkedList<>();
-        this.ignoredSchemaChanges = new LinkedList<>();
         this.schemaManager = schemaManager;
         this.schemaDerivation = schemaDerivation;
+        this.schemaChangeBehavior = schemaChangeBehavior;
+
+        this.activeSinkWriters = new HashSet<>();
+        this.flushedSinkWriters = new HashSet<>();
         this.schemaChangeThreadPool = Executors.newSingleThreadExecutor();
+
         this.isSchemaChangeApplying = false;
-        this.schemaChangeBehavior = schemaChangeBehavior;
+        this.currentDerivedSchemaChangeEvents = new ArrayList<>();
+        this.currentFinishedSchemaChanges = new ArrayList<>();
+        this.currentIgnoredSchemaChanges = new ArrayList<>();
+        this.isSchemaChangeRequested = new AtomicBoolean(false);
+    }
+
+    /**
+     * Handle the {@link SchemaChangeRequest} and wait for all sink subtasks flushing.
+     *
+     * @param request the received SchemaChangeRequest
+     */
+    public CompletableFuture<CoordinationResponse> handleSchemaChangeRequest(
+            SchemaChangeRequest request) {
+        if (isSchemaChangeRequested.compareAndSet(false, true)) {
+            isSchemaChangeApplying = true;
+            LOG.info(
+                    "Received schema change event request {} from table {}. Start to buffer requests for others.",
+                    request.getSchemaChangeEvent(),
+                    request.getTableId().toString());
+            SchemaChangeEvent event = request.getSchemaChangeEvent();
+            if (schemaManager.isOriginalSchemaChangeEventRedundant(event)) {
+                LOG.info("Event {} has been addressed before, ignoring it.", event);
+                finishCurrentSchemaChangeRequest();
+                return CompletableFuture.completedFuture(
+                        wrap(new SchemaChangeResponse(Collections.emptyList())));
```

Comment:
Could we add some message in SchemaChangeResponse body to let clients know more details? it will be helpful for troubleshooting